tlatools/src/tlc2/tool/Simulator.java

```diff
@@ -103,41 +114,303 @@ public Simulator(String specFile, String configFile, String traceFile, boolean d
 	private final Action[] impliedActions; // the implied-actions to be checked
 	private final boolean checkDeadlock; // check deadlock?
 	private final boolean checkLiveness; // check liveness?
-	private long numOfGenStates;
+	
+	// The total number of states generated by all workers. Written to concurrently.
+	private AtomicLong numOfGenStates;
```

Comment:
[LongAdder](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/LongAdder.html) to reduce contention?

Reply:
Done.